src/components/[guild]/Requirements/components/GuildCheckout/components/PaymentCurrencyPicker/components/TokenInfo.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  Circle,
+  HStack,
+  Img,
+  MenuItem,
+  Skeleton,
+  SkeletonCircle,
+  Stack,
+  Text,
+  useColorModeValue,
+} from "@chakra-ui/react"
+import { formatUnits } from "@ethersproject/units"
+import usePrice from "components/[guild]/Requirements/components/GuildCheckout/hooks/usePrice"
+import { Chains, RPC } from "connectors"
+import useCoinBalance from "hooks/useCoinBalance"
+import useTokenBalance from "hooks/useTokenBalance"
+import useTokenData from "hooks/useTokenData"
+import { Fragment } from "react"
+import { Rest } from "types"
+
+type Props = {
+  chainId: number
+  address: string
+  asMenuItem?: boolean
+} & Rest
+
+const ADDRESS_REGEX = /^0x[A-F0-9]{40}$/i
+
+const TokenInfo = ({
+  chainId,
+  address,
+  asMenuItem,
+  ...rest
+}: Props): JSX.Element => {
+  const circleBgColor = useColorModeValue("blackAlpha.100", "blackAlpha.300")
+
+  const {
+    data: { priceInSellToken },
+    isValidating: isPriceDataLoading,
+    error: priceError,
+  } = usePrice(address)
+
+  const isTooSmallPrice = priceInSellToken
+    ? parseFloat(priceInSellToken.toFixed(3)) <= 0.0
+    : undefined
+
+  const {
+    data: { symbol: tokenSymbol, decimals, logoURI: tokenLogoURI },
+    error,
+    isValidating: isTokenDataLoading,
+  } = useTokenData(Chains[chainId], address)
+  const logoURI = ADDRESS_REGEX.test(address)
+    ? tokenLogoURI
+    : RPC[Chains[chainId]].nativeCurrency.logoURI
+  const symbol = ADDRESS_REGEX.test(address)
+    ? tokenSymbol
+    : RPC[Chains[chainId]].nativeCurrency.symbol
+
+  const { balance: coinBalance, isLoading: isCoinBalanceLoading } =
+    useCoinBalance(chainId)
+  const { balance: tokenBalance, isLoading: isTokenBalanceLoading } =
+    useTokenBalance(address, chainId)
+  const balance = formatUnits(
+    (address === RPC[Chains[chainId]]?.nativeCurrency?.symbol
+      ? coinBalance
+      : tokenBalance) ?? "0",
+    address === RPC[Chains[chainId]]?.nativeCurrency?.symbol
+      ? RPC[Chains[chainId]]?.nativeCurrency?.decimals
+      : decimals ?? 18
+  )
```

Comment:
It would be nice if we could merge useCoinBalance & useTokenBalance into one hook that works for both (sidequest as per boy scout rule)